File: packages/persister/src/influx.ts
import { InfluxDB } from "@influxdata/influxdb-client";

import { requireEnv } from "./utils";

export const createInfluxClient = () => {
  const client = new InfluxDB({
    url: requireEnv("INFLUX_URL"),
    token: requireEnv("INFLUX_TOKEN"),
  });

  const write = client.getWriteApi(
    requireEnv("INFLUX_ORG"),
    requireEnv("INFLUX_BUCKET"),
  );

  const query = client.getQueryApi(requireEnv("INFLUX_ORG"));

  return { client, write, query };
};

File: packages/api/server/src/main/kotlin/config/Influx.kt
package io.t8n.kedeng.config

import com.influxdb.client.InfluxDBClientOptions

fun influxConfig() = InfluxDBClientOptions.builder()
    .url(Environment.get("INFLUX_URL"))
    .bucket(Environment.get("INFLUX_BUCKET"))
    .org(Environment.get("INFLUX_ORG"))
    .authenticateToken(Environment.get("INFLUX_TOKEN").toCharArray())
    .build()